console/prompter.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package console
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/peterh/liner"
+)
+
+// Stdin holds the stdin line reader (also using stdout for printing prompts).
+// Only this reader may be used for input because it keeps an internal buffer.
+var Stdin = newTerminalPrompter()
+
+// UserPrompter defines the methods needed by the console to promt the user for
+// various types of inputs.
+type UserPrompter interface {
+	// PromptInput displays the given prompt to the user and requests some textual
+	// data to be entered, returning the input of the user.
+	PromptInput(prompt string) (string, error)
+
+	// PromptPassword displays the given prompt to the user and requests some textual
+	// data to be entered, but one which must not be echoed out into the terminal.
+	// The method returns the input provided by the user.
+	PromptPassword(prompt string) (string, error)
+
+	// PromptConfirm displays the given prompt to the user and requests a boolean
+	// choice to be made, returning that choice.
+	PromptConfirm(prompt string) (bool, error)
+
+	// SetScrollHistory sets the the input scrollback history that the prompter will
+	// allow the user to scoll back to.
+	SetScrollHistory(history []string)
```

Comment:
It feels weird to call this "scroll history". Why can't it just be `SetHistory`, `AppendHistory`?
This should not prevent merge but please feel free to change these names if you agree.